ra2ce/graph/hazard.py

```diff
@@ -887,88 +924,112 @@ def create(self):
+        if self.files["base_network"]:
+            if self.files["base_network_hazard"] is None:
+                # Check if the graph needs to be reprojected
+                hazard_crs = pyproj.CRS.from_user_input(
+                    self.config["hazard"]["hazard_crs"]
+                )
+                gdf_crs = pyproj.CRS.from_user_input(self.graphs["base_network"].crs)
 
-            if (
-                hazard_crs != gdf_crs
-            ):  # Temporarily reproject the graph to the CRS of the hazard
-                logging.warning(
-                    """Hazard crs {} and gdf crs {} are inconsistent,
-                                                we try to reproject the gdf crs""".format(
-                        hazard_crs, gdf_crs
+                if (
+                    hazard_crs != gdf_crs
+                ):  # Temporarily reproject the graph to the CRS of the hazard
+                    logging.warning(
+                        """Hazard crs {} and gdf crs {} are inconsistent,
+                                                    we try to reproject the gdf crs""".format(
+                            hazard_crs, gdf_crs
+                        )
+                    )
+                    extent_gdf = self.graphs["base_network"].total_bounds
+                    logging.info(
+                        "Gdf extent before reprojecting: {}".format(extent_gdf)
+                    )
+                    gdf_reprojected = (
+                        self.graphs["base_network"].copy().to_crs(hazard_crs)
+                    )
+                    extent_gdf_reprojected = gdf_reprojected.total_bounds
+                    logging.info(
+                        "Gdf extent after reprojecting: {}".format(
+                            extent_gdf_reprojected
+                        )
                     )
-                )
-                extent_gdf = self.graphs["base_network"].total_bounds
-                logging.info("Gdf extent before reprojecting: {}".format(extent_gdf))
-                gdf_reprojected = self.graphs["base_network"].copy().to_crs(hazard_crs)
-                extent_gdf_reprojected = gdf_reprojected.total_bounds
-                logging.info(
-                    "Gdf extent after reprojecting: {}".format(extent_gdf_reprojected)
-                )
 
-                # Do the actual hazard intersect
-                gdf_reprojected = self.hazard_intersect(gdf_reprojected)
+                    # Do the actual hazard intersect
+                    gdf_reprojected = self.hazard_intersect(gdf_reprojected)
+
+                    # Assign the original geometries to the reprojected raster
+                    original_geometries = self.graphs["base_network"]["geometry"]
+                    gdf_reprojected["geometry"] = original_geometries
+                    self.graphs["base_network_hazard"] = gdf_reprojected.copy()
+                    del gdf_reprojected
```

Comment:
why not assigning without copying instead of copying and deleting? I see this as a waste of memory :/